WireAPI/Sources/WireAPI/APIs/UpdateEventsAPI/Event decoding/UpdateEventDecodingError.swift

```diff
@@ -0,0 +1,39 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+/// An error when decoding `UpdateEvent`.
+
+public struct UpdateEventDecodingError: Error, CustomStringConvertible {
+
+    /// The type of the event being decoding.
+
+    public let eventType: String
+
+    /// The error that occurred when decoding.
+
+    public let decodingError: Error
```

Comment:
`UpdateEventDecodingError` keeps track of both the error that occurred when decoding an event, as well as the event type that couldn't be decoding. We don't catch the specific error but catch all errors and log them. I found that that the decoding error is quite descriptive, but since there are many event types, it still not enough information to know how to use the error to find the issue.